### ISSUE TITLE
Query format | Values intended to be queried by tag 'epoch'  not queried when invoked by tag 'evr'

### ISSUE THREAD

**Issue by Ricky-Tigg** [rpm-software-management/rpm#2364]
Hello. Tag _evr_ fails to query the values intended to be queried by tag _epoch_.
```
$ rpm -q --qf %{evr} rpm
4.18.0-1.fc37
```
Though the underlying function for tag _epoch_ behaves as intended.
```
$ rpm -q --qf %{epoch} rpm
(none)
```
Though _0_ instead of _(none)_ could be a bit more adequate mention.

**Comment by Ricky-Tigg** [comment]
Issue valid with tag _nevra_.

**Comment by pmatilai** [comment]
No, this is expected behavior. That package does not HAVE an epoch, so it's not reported. So while zero epoch equals no epoch semantically, rpm does show the difference.

**Comment by Ricky-Tigg** [comment]
"_That package does not HAVE an epoch_"

Wasn't it explicit from my output resulting `rpm -q --qf %{epoch} rpm`?

"_That package does not HAVE an epoch, so it's not reported._"

Well, not reported! Was there something that prevented you from reading a _(none)_ from my output resulting `rpm -q --qf %{epoch} rpm`? Do you read it at last? It seems obvious to me that in the present context, a coherent flow would have to lead to produce as follows:
```
$ rpm -q --qf %{evr} rpm
(none):4.18.0-1.fc37
```
An you however, you did manage to close. Can anyone find any coherence in such action that could benefit the present project?

**Comment by Ricky-Tigg** [comment]
"_That package does not HAVE an epoch_"

Wasn't it explicit from my output resulting `rpm -q --qf %{epoch} rpm`?

"_That package does not HAVE an epoch, so it's not reported._"

Well, not reported! Was there something that prevented you from reading a _(none)_ from my output resulting `rpm -q --qf %{epoch} rpm`? Do you read it at last? It seems obvious to me that in the present context, a coherent flow would have to lead to produce as follows:
```
$ rpm -q --qf %{evr} rpm
(none):4.18.0-1.fc37
```
And you however, you did manage to close. Can anyone find any coherence in a such action that could benefit the present project?

**Comment by pmatilai** [comment]
I did read your entire report. As said, this is how rpm is expected to behave. That may not match what *your* expectations, but that doesn't mean it's a bug.

**Comment by Ricky-Tigg** [comment]
There cannot be better model for illustrating a bad use of convention, since such a convention could not be determined invariably as a convention.

**Proof-case**
Is assumed:
- a **non-null** _epoch_ value exits for an installed package
- user not aware of the exitence of a **non-null** _epoch_ value for an installed package
- **non-null** _epoch_ values not reported by `rpm -q --qf %{evr} <package>` nor any other mention reported

Then the very failure of reporting anything could not be determined either as a consequence of a convention or as an issue.

**Comment by Ricky-Tigg** [comment]
There cannot be better model for illustrating a bad use of convention, since such a convention could not be determined invariably as a convention.

**Proof-case**
Is assumed:
- a **non-null** _epoch_ value exits for an installed package
- user not aware of the existence of a **non-null** _epoch_ value for an installed package
- **non-null** _epoch_ values not reported by `rpm -q --qf %{evr} <package>` nor any other mention reported

Then the very failure of reporting anything could not be determined either as a consequence of a convention or as an issue.

**Comment by dralley** [comment]
"evr" is a defined strategy for creating a string summarizing a package's critical details as a string, which the user can query.  `(none)` is just a special value the `rpm` cli uses to represent the nonexistence of a tag.  I would not expect an EVR to ever look like `(none):4.18.0-1.fc37`, that would be leaking details about how information is displayed in the CLI backwards into the tag query mechanism.

In other words, when you query the "evr" you're not querying the epoch, version and release and asking the CLI to display them.  You're asking the CLI to display the "evr" value as generated by `rpm`.  And there is no reason for that to ever contain `(none)`.

**Comment by Ricky-Tigg** [comment]
So that is all what it was about; **deliberate inconsistency.** Choosing to print sometimes _epoch_ values, with tag _epoch_, and sometimes not to print them, with tags _evr_ and _nevra_, would not be expected from developers. Having such a fantasy in your code must have pleased you so far since you showed interest to keep it. After all, that makes the motive for suddenly closing enterely different, and it was indeed worth being closed as there was nothing that could have been done here in such a context.

**Comment by dralley** [comment]
It's not a matter of "sometimes printing them and sometimes not" because you're querying two different things.  Querying the "epoch" is not the same as querying the "evr".  

Really the only argument that can be made is that for nonexistent tags, rpm-cli should print nothing instead of `(none)`, which I suppose is debatable, but backwards compatibility isn't going to be broken over something so trivial.
